clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -245,12 +247,23 @@ public int partitionLeaderEpoch() {
             protected Record readNext(long baseOffset, long baseTimestamp, int baseSequence, Long logAppendTime) {
                 try {
                     return DefaultRecord.readFrom(inputStream, baseOffset, baseTimestamp, baseSequence, logAppendTime);
+                } catch (EOFException e) {
+                    throw new InvalidRecordException("Incorrect declared batch size, premature EOF reached");
                 } catch (IOException e) {
                     throw new KafkaException("Failed to decompress record stream", e);
                 }
             }
 
             @Override
+            protected boolean hasRemaining() {
+                try {
+                    return inputStream.read() == -1;
```

Comment:
@ijuma right, right, sorry about that. But it was you who suggested narrowing the try block. Care to comment below?